main.go

```diff
@@ -292,7 +299,12 @@ func parseConnData(r *http.Request) (connData, error) {
 
 	sha := sha1.New()
 	fmt.Fprint(sha, service)
-	fmt.Fprint(sha, category)
+	// don't break anything that's depending on the old format of category
+	if len(categories) == 1 {
+		fmt.Fprint(sha, categories[0])
+	} else {
+		fmt.Fprint(sha, categories)
```

Comment:
This should probably be `fmt.Fprintf(sha, "#%v", categories)` to prevent weird edge cases. For example:

```
fmt.Println([]string{"foo","bar"})
fmt.Println([]string{"foo bar"})
```

Both output the same thing. In that light the other print statements that are writing to `sha` probably need similar protection but that can wait for another time.